### PR TITLE
Upgrade build Kotlin version to 1.6.20 GA

### PR DESCRIPTION
**Pull request by blindpirate**

### CODE REVIEW

**Comment by blindpirate** [comment]
@bot-gradle test and merge

**Comment by blindpirate** [comment]
@bot-gradle cancel

**Comment by blindpirate** [comment]
@bot-gradle test this

File: subprojects/distributions-dependencies/build.gradle.kts [hunk 1]
```diff
@@ -129,7 +129,7 @@ dependencies {
         api(libs.snakeyaml)             { version { strictly("1.28") }}
         api(libs.testng)                { version { strictly("6.3.1"); because("later versions break test cross-version test filtering") }}
         api(libs.tomlj)                 { version { strictly(tomljVersion) }}
-        api(libs.trove4j)               { version { strictly("1.0.20181211") }}
+        api(libs.trove4j)               { version { strictly("1.0.20200330") }}
```
Comment (wolfs) [inline subprojects/distributions-dependencies/build.gradle.kts:132]: Interesting, does this need to be in sync with what the Kotlin plugin uses?

File: subprojects/smoke-test/src/smokeTest/groovy/org/gradle/smoketests/KotlinPluginSmokeTest.groovy [hunk 2]
```diff
@@ -312,7 +312,7 @@ class KotlinPluginSmokeTest extends AbstractPluginValidatingSmokeTest implements
         void expectKotlinParallelTasksDeprecation(String version) {
             VersionNumber versionNumber = VersionNumber.parse(version)
             runner.expectLegacyDeprecationWarningIf(
-                versionNumber >= VersionNumber.parse('1.5.20') && versionNumber <= VersionNumber.parse('1.6.10'),
+                versionNumber >= VersionNumber.parse('1.5.20') && versionNumber <= VersionNumber.parse('1.6.20'),
```
Comment (wolfs) [inline subprojects/smoke-test/src/smokeTest/groovy/org/gradle/smoketests/KotlinPluginSmokeTest.groovy:315]: I'd expect this deprecation warning to not show up with 1.6.20.

File: subprojects/smoke-test/src/smokeTest/groovy/org/gradle/smoketests/WithKotlinDeprecations.groovy [hunk 3]
```diff
@@ -41,7 +41,7 @@ trait WithKotlinDeprecations {
     void expectKotlinCompileDestinationDirPropertyDeprecation(String version) {
         VersionNumber versionNumber = VersionNumber.parse(version)
         runner.expectLegacyDeprecationWarningIf(
-            versionNumber >= VersionNumber.parse('1.5.20') && versionNumber <= VersionNumber.parse('1.6.10'),
+            versionNumber >= VersionNumber.parse('1.5.20') && versionNumber <= VersionNumber.parse('1.6.20'),
```
Comment (wolfs) [inline subprojects/smoke-test/src/smokeTest/groovy/org/gradle/smoketests/WithKotlinDeprecations.groovy:44]: I'd expect this deprecation warning to not show up with 1.6.20.

**Review by wolfs — APPROVED** [review]
I think this looks good, I left some comments.

File: subprojects/distributions-dependencies/build.gradle.kts (hunk 1 above)
Comment (eskatos) [reply subprojects/distributions-dependencies/build.gradle.kts:132]: It needs to be aligned with the embedded kotlin compiler.

**Comment by blindpirate** [comment]
@bot-gradle test this

**Comment by blindpirate** [comment]
@bot-gradle test this

**Comment by blindpirate** [comment]
@bot-gradle test and merge

**Comment by blindpirate** [comment]
@bot-gradle test and merge

**Review by eskatos — APPROVED** [review]
:+1:

File: gradle/verification-metadata.xml [hunk 4]
```diff
@@ -1258,71 +1258,16 @@
...
-         </artifact>
-      </component>
       <component group="org.gradle.kotlin" name="gradle-kotlin-dsl-plugins" version="2.2.0">
          <artifact name="gradle-kotlin-dsl-plugins-2.2.0.jar">
             <sha256 value="21acc544c399c1b9dd7f1c2a6c5eacec8f8ceab5a0ad21e064c172d3ee288b89" origin="Artifact is not signed"/>
          </artifact>
       </component>
+      <component group="org.gradle.kotlin" name="gradle-kotlin-dsl-plugins" version="2.3.0">
+         <artifact name="gradle-kotlin-dsl-plugins-2.3.0.jar">
+            <sha256 value="b14c5993e4b97b85aa0696ec7edb744021bc82795317d9cafa662c5fc9d9f590" origin="Generated by Gradle"/>
```
Comment (wolfs) [inline gradle/verification-metadata.xml:1268]: We should probably set up signing for this plugin as well.

**Review by wolfs — APPROVED** [review]
LGTM!

**Comment by blindpirate** [comment]
@bot-gradle test and merge

**Comment by blindpirate** [comment]
@bot-gradle test and merge

**Comment by blindpirate** [comment]
@bot-gradle test and merge

**Comment by blindpirate** [comment]
@bot-gradle test and merge

**Comment by blindpirate** [comment]
@bot-gradle test and merge

**Pull request merged**